shared/chat/inbox/container/remote.js

```diff
@@ -26,47 +17,56 @@ export type RemoteConvMeta = $Diff<
   {onSelectConversation: () => void}
 >
 
-// Sort by timestamp
-const getSortedConvMetas = Container.createSelector([getMetas], ([map, state]) =>
-  map
-    .sort((a, b) => b.timestamp - a.timestamp)
-    .slice(0, maxShownConversations)
-    .toList()
-    .map((m): RemoteConvMeta => {
-      const hasUnread = Constants.getHasUnread(state, m.conversationIDKey)
-      const styles = Constants.getRowStyles(m, false, hasUnread)
-      const participantNeedToRekey = m.rekeyers.size > 0
-      const _username = state.config.username || ''
-      const youNeedToRekey = !!participantNeedToRekey && m.rekeyers.has(_username)
-      return {
-        backgroundColor: Styles.globalColors.white,
-        channelname: m.channelname,
-        conversationIDKey: m.conversationIDKey,
-        hasBadge: Constants.getHasBadge(state, m.conversationIDKey),
-        hasResetUsers: !!m.resetParticipants && m.resetParticipants.size > 0,
-        hasUnread,
-        iconHoverColor: styles.iconHoverColor,
-        isFinalized: !!m.wasFinalizedBy,
-        isInWidget: true,
-        isMuted: m.isMuted,
-        isSelected: false,
-        // excluding onSelectConversation
-        participantNeedToRekey,
-        participants: m.teamname ? [] : Constants.getRowParticipants(m, _username).toArray(),
-        showBold: styles.showBold,
-        snippet: m.snippet,
-        snippetDecoration: m.snippetDecoration,
-        subColor: styles.subColor,
-        teamname: m.teamname,
-        timestamp: Constants.timestampToString(m),
-        usernameColor: styles.usernameColor,
-        youAreReset: m.membershipType === 'youAreReset',
-        youNeedToRekey,
-      }
-    }).toArray()
+// To cache the list
+const valuesCached = memoize((...vals) => vals.map(v => v))
+
+const metaMapToFirstValues = memoize(metaMap =>
+  metaMap
+    .partialSort(maxShownConversations, (a, b) => b.timestamp - a.timestamp)
```

Comment:
Why did you remove the filter? We're going to need to add to it anyway, so please leave it in.

Reply:
Also where is `partialSort` coming from? I don't see it in Immutable's Map, just `immutable-sorted`, which we don't seem to be using.

Reply:
Nvmind, we're using `immutable-sorted` in our `package.json` as the `immutable` package.